component/config.go

```diff
@@ -21,12 +21,29 @@ import (
 // Type is the component type as it is used in the config.
 type Type string
 
-// validatable defines the interface for the configuration validation.
-type validatable interface {
-	// Validate validates the configuration and returns an error if invalid.
+type Config interface {
+	identifiable
+
+	privateConfig()
+}
+
+// ConfigValidator defines an optional interface for configurations to implement to do validation.
+type ConfigValidator interface {
```

Comment:
Why make this optional? I thought we wanted to ensure every component implemented this, otherwise, as we saw in contrib, each component implemented validated in a slightly different way, and some components did not implement this at all.

Reply:
Right now is optional anyway since we are implementing it in the "*Settings", so only some components override it. This PR does not change that it any way, it changes only on how the "optionality" is implemented.

If we want to move to enforce "Validate" it is fine, but will be a big breaking change I think. Happy to do that way as well, as long as we no longer rely on "Settings" to have the "default" implementation for this func.

Reply:
Any thoughts on how big of a breaking change? I recall adding Validate to *Settings caused a big breaking change because so many components didn't have a Validate func. I'd be curious if there are many more left that do not.

What will be the way to replace *Settings? Is it to embed `component.ID` into the specific component struct directly?

Reply:
> Any thoughts on how big of a breaking change? I recall adding Validate to *Settings caused a big breaking change because so many components didn't have a Validate func. I'd be curious if there are many more left that do not.

Let me google(grep) that for you:

```bash
➜  opentelemetry-collector-contrib git:(main) grep -R "type Config struct" | wc -l
     212
➜  opentelemetry-collector-contrib git:(main) grep -R "func (.*Config) Validate() error" | wc -l 
     106
```

> What will be the way to replace *Settings? Is it to embed component.ID into the specific component struct directly?

`config.*Settings` are used for 2 reasons (you can also see in the code):
* Implement the `identifiable` interface to avoid duplicate code of implementing that.
* Implement the `validatable` interface to not force everyone to have to implement it.
  
For the `validatable` this is the proposed solution; For `identifiable` I actually think to not pass the ID via the Config between the Host/Service and the `Component` but to pass it via the `CreateSettings` (see https://github.com/open-telemetry/opentelemetry-collector/blob/main/component/receiver.go#L136)

Reply:
That note was resolved offline, because the mapping (ID -> Config) still exists in the service.Config.